### PR TITLE
[RFC] base: ostree: add support systemd-boot automatic boot assesment

### PR DESCRIPTION
**Pull request by igoropaniuk**
Add support for Automatic Boot Assessment [1].
Boot entries are now created with an additional suffix, which represents the amount of maximum tries for boot counting.

[1] https://systemd.io/AUTOMATIC_BOOT_ASSESSMENT/
Signed-off-by: Igor Opaniuk <igor.opaniuk@foundries.io>

### CODE REVIEW

File: meta-lmp-base/recipes-extended/ostree/ostree/0006-Add-support-systemd-boot-automatic-boot-assesment.patch [hunk 1]
```diff
@@ -0,0 +1,148 @@
+From a9e35b76fd5e47b9bc29a7113d97a4d98a973e75 Mon Sep 17 00:00:00 2001
+From: Igor Opaniuk <igor.opaniuk@foundries.io>
+Date: Wed, 11 Sep 2024 18:03:10 +0200
+Subject: [PATCH] Add support systemd-boot automatic boot assesment
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-extended/ostree/ostree/0006-Add-support-systemd-boot-automatic-boot-assesment.patch:4]: s/assesment/assessment/

**Comment by ricardosalveti** [comment]
Change itself looks ok, is bless boot enabled by default? Just trying to understand if once we land this patch everything will keep working the same way, until we move the confirmation to aktualizr-lite or similar.

**Comment by igoropaniuk** [comment]
> is bless boot enabled by default

yes, I've tested in on `intel-corei7-64`, systemd service configurations (`systemd-bless-boot.service`, `boot-complete.target`) and tools are installed to rootfs by default (`systemd-bless-boot`)

Systemd-boot EFI app handles all `+n` suffixes in boot entries automatically (no need to enable anything explicitly )

File: meta-lmp-base/recipes-extended/ostree/ostree/0006-Add-support-systemd-boot-automatic-boot-assesment.patch [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+Add support for Automatic Boot Assessment [1].
+Boot entries are now created with an additional suffix, which
+represents the amount of maximum tries for boot counting.
+
+[1] https://systemd.io/AUTOMATIC_BOOT_ASSESSMENT/
+Upstream-Status: Inappropriate [lmp specific]
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-extended/ostree/ostree/0006-Add-support-systemd-boot-automatic-boot-assesment.patch:11]: We should probably set as pending here, as this is something we want to try to upstream once there is support for systemd.

**Comment by quic-roshs** [comment]
Shouldn't we be checking if the bootloader used is systemd-boot before appending the counter to the conf file? 
This might be needed if we are planning to upstream the patch.

**Comment by igoropaniuk** [comment]
@quic-roshs this behavior is described in Bootloader Specification (https://uapi-group.org/specifications/specs/boot_loader_specification/#boot-counting; as well as OSTree also follows it https://ostreedev.github.io/ostree/deployment/#the-system-boot), so this is not systemd-boot specific adjustments

**Comment by ricardosalveti** [comment]
> @quic-roshs this behavior is described in Bootloader Specification (https://uapi-group.org/specifications/specs/boot_loader_specification/#boot-counting; as well as OSTree also follows it https://ostreedev.github.io/ostree/deployment/#the-system-boot), so this is not systemd-boot specific adjustments

Is u-boot (uEnv) based boot flows reusing the same code path?

**Comment by igoropaniuk** [comment]
@ricardosalveti 

> Is u-boot (uEnv) based boot flows reusing the same code path?

yes, to create the final uEnv.txt it parses boot entries that was created before by the function I adjusted in this PR with `create_config_from_boot_loader_entries()` function (https://github.com/ostreedev/ostree/blob/main/src/libostree/ostree-bootloader-uboot.c#L102) .
The change of boot entries naming doesn't somehow affect the contents of the final `uEnv.txt`

**Comment by ricardosalveti** [comment]
Needs a rebase.

**Comment by quaresmajose** [comment]
merged manually

**Pull request closed without merging**